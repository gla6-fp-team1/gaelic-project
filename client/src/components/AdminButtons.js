const AdminButtons = ({ user }) => {
	const handleExportGaelicData = async () => {
		const link = document.createElement("a");

		link.href = "/api/sentences/export";
		link.download = "exported_data.json";

		document.body.appendChild(link);
		link.click();

		setTimeout(() => {
			document.body.removeChild(link);
		}, 0);
	};

	return (
		<>
			{user && user.permissions && user.permissions.isAdmin && (
				<div className="adminFunctions">
					<p> Admin Functions:</p>
					<div className="fileUpload">
						<form
							method="POST"
							action="/api/sentences/upload"
							encType="multipart/form-data"
						>
							<input type="file" name="file" id="fileInput" />
							<input type="submit" value="Upload File" />
						</form>
					</div>
					<button id="myGaelicButton" onClick={handleExportGaelicData}>
						ExportGaelicData
					</button>
				</div>
			)}
		</>
	);
};
export default AdminButtons;
