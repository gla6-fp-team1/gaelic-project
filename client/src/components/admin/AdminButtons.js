const AdminButtons = () => {
	const handleExportDatabase = async () => {
		const link = document.createElement("a");

		link.href = "/api/sentences/export";
		link.download = "exported_data.json";

		document.body.appendChild(link);
		link.click();

		setTimeout(() => {
			document.body.removeChild(link);
		}, 0);
	};

	return (
		<>
			<div className="file-upload">
				<h2>Upload sentence data:</h2>
				<form
					method="POST"
					action="/api/sentences/upload"
					encType="multipart/form-data"
				>
					<input type="file" name="file" id="fileInput" />
					<input type="submit" value="Upload File" />
				</form>
			</div>
			<div className="export-database">
				<h2>Export database to JSON file:</h2>
				<button onClick={handleExportDatabase}>Export Database</button>
			</div>
		</>
	);
};
export default AdminButtons;
