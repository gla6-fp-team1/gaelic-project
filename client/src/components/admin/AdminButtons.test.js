import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AdminButtons from "./AdminButtons";

it("Renders the admin buttons table", async () => {
	const { asFragment } = render(<AdminButtons />);
	expect(asFragment()).toMatchSnapshot();
});
