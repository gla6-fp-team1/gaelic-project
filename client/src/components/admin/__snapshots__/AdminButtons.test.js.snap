// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the admin buttons table 1`] = `
<DocumentFragment>
  <div
    class="file-upload"
  >
    <h2>
      Upload sentence data:
    </h2>
    <form
      action="/api/sentences/upload"
      enctype="multipart/form-data"
      method="POST"
    >
      <input
        id="fileInput"
        name="file"
        type="file"
      />
      <input
        type="submit"
        value="Upload File"
      />
    </form>
  </div>
  <div
    class="export-database"
  >
    <h2>
      Export database to JSON file:
    </h2>
    <button>
      Export Database
    </button>
  </div>
</DocumentFragment>
`;
